pkg/plugins/runtime/gateway/generator.go

```diff
@@ -62,6 +64,7 @@ type GatewayListener struct {
 	// CrossMesh is important because for generation we need to treat such a
 	// listener as if we have HTTPS with the Mesh cert for this Dataplane
 	CrossMesh bool
+	Resources *mesh_proto.MeshGateway_Listener_Resources // TODO verify these don't conflict when merging
```

Comment:
will this be a part of this PR? a separate issue?

Reply:
Good catch, this PR for sure, i just want to get consensus on the whether we're going this route.

Reply:
@jakubdyszkiewicz done